tests/rptest/tests/cluster_metadata_upload_test.py

```diff
@@ -0,0 +1,136 @@
+    (that would be used in a recovery) only references objects that exist in
+    the bucket.
+    """
+    latest_cluster_metadata = ClusterMetadata(dict(), dict())
+    highest_meta_id = -1
+    cluster_uuid = ""
+    for uuid, meta in s3_snapshot.cluster_metadata.items():
+        for meta_id, _ in meta.cluster_metadata_manifests.items():
+            if meta_id > highest_meta_id:
+                latest_cluster_metadata = meta
+                cluster_uuid = uuid
+                highest_meta_id = meta_id
+    highest_manifest = latest_cluster_metadata.cluster_metadata_manifests[
+        highest_meta_id]
+    assert 'controller_snapshot_path' in highest_manifest, highest_manifest
+    controller_snapshot_path = highest_manifest['controller_snapshot_path']
+    assert len(controller_snapshot_path) > 0, highest_manifest
+
+    _, snap_offset = parse_controller_snapshot_path(controller_snapshot_path)
+    assert snap_offset in latest_cluster_metadata.controller_snapshot_sizes, \
+        f"Couldn't find {snap_offset} in {latest_cluster_metadata.controller_snapshot_sizes}"
+    assert latest_cluster_metadata.controller_snapshot_sizes[snap_offset] > 0
+    return cluster_uuid, highest_meta_id
+
+
+class ClusterMetadataUploadTest(RedpandaTest):
+    topic_name = "oolong"
+    topics = (TopicSpec(name=topic_name, partition_count=10), )
+    segment_size = 1024 * 1024
+
+    def __init__(self, test_context):
+        si_settings = SISettings(test_context,
+                                 log_segment_size=self.segment_size,
+                                 fast_uploads=True)
+        extra_rp_conf = {
+            'controller_snapshot_max_age_sec': 1,
+            'cloud_storage_cluster_metadata_upload_interval_ms': 1000,
+            'disable_cluster_metadata_upload_loop': False,
+        }
+        super().__init__(test_context,
+                         si_settings=si_settings,
+                         num_brokers=3,
+                         extra_rp_conf=extra_rp_conf)
+        self.redpanda.start_si()
+
+    def bucket_has_metadata(self, at_least: int):
+        """
+        Returns true if at least some metadata exist in the bucket.
+        As new metadata is written, the upload loop may be cleaning up after
+        itself, so ignores exceptions.
+        """
+        try:
+            s3_snapshot = BucketView(self.redpanda, topics=self.topics)
+            num_cluster_metas = 0
+            for _, meta in s3_snapshot.cluster_metadata.items():
+                if len(meta.cluster_metadata_manifests) > 0 and len(
+                        meta.controller_snapshot_sizes) > 0:
+                    num_cluster_metas += 1
+            return num_cluster_metas >= at_least
+        except Exception as e:
+            self.logger.warn(f"Error while populating snapshot: {str(e)}")
+            return False
+
+    @cluster(num_nodes=3)
+    def test_uploads_after_restart(self):
```

Comment:
You need the following incantation to get your test to run on both azurite and minio
```
@cluster(num_nodes=3)
@matrix(cloud_storage_type=get_cloud_storage_type())
def  def test_uploads_after_restart(self, cloud_storage_type):
```

Reply:
Done